tests/python/unittest/test_operator.py

```diff
@@ -6977,6 +6977,109 @@ def test_roi_align_autograd(sampling_ratio=0):
     test_roi_align_value(2)
     test_roi_align_autograd()
 
+@with_seed()
+def test_diag():
+
+    # test 2d input
+    h = np.random.randint(2,9)
+    w = np.random.randint(2,9)
+    a_np = np.random.random((h, w))
+    a = mx.nd.array(a_np)
+    
+    # k == 0
+    r = mx.nd.diag(a)
+
+    for i in range(r.shape[0]):
+        assert r[i] == a[i][i]
```

Comment:
you can use numpy for consistency check. It's more stable than manually setting the desired values

Reply:
Good point, will do

Reply:
Incorporated the changes.